### PR TITLE
Enforced newline-per-chained-call

### LINKED ISSUE #1019: [DevEx]: Configure eslint rule: newline-per-chained-call
### Improvement Summary

In order to reduce stylistic discrepancies between contributors regarding line length, we should apply the newline-per-chained-call eslint rule

### Detailed Description

Here is the rule definition: https://eslint.style/rules/js/newline-per-chained-call. We should use the default value of 2 for `ignoreChainWithDepth` to allow short chains of 2 commands to use a single line. We should run `npm run lint:fix` and commit the changes after configuring the rule to normali3 line length across the project.

This may require installing an additional dev dependency for style rules.

### PR DESCRIPTION
**Pull request by Mang0codes**
## Issue number

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #1019

## Please check the following

- [ ] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

### CODE REVIEW

**Comment by Mang0codes** [comment]
@itsalaidbacklife I closed that pull request and also created a PR for the LF issue you raised. Is this okay now?

File: api/helpers/ find-spectatable-games.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ module.exports = {
   description: 'Finds all the games that are available to spectate',
 
   fn: async (_, exits) => {
-    const recentUpdateThreshhold = dayjs.utc().subtract(5, 'minute').toDate();
+    const recentUpdateThreshhold = dayjs.utc().subtract(5, 'minute')
+.toDate();
```
Comment (seriouslysean) [inline api/helpers/ find-spectatable-games.js:12]: I would expect these to indent properly, and not just be flush far left
```
    const recentUpdateThreshhold = dayjs.utc().subtract(5, 'minute')
        .toDate();
```

File: .eslintrc.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ module.exports = {
         }
       }
     ],
+    'newline-per-chained-call': ['error', { ignoreChainWithDepth: 2 }],
```
Comment (seriouslysean) [inline .eslintrc.js:58]: It seems like you may also need something like this to enforce proper indenting, but you'd need to test it.
```
'indent': ['error', 2],
```

Comment (Mang0codes) [reply .eslintrc.js:58]: Yes, the indentation seemed odd to me but adding this `'indent': ['error', 2],` will affect the entire codebase which means some of the existing codes will be changed too. If you're okay with this I will commit the changes.
Screenshot 2024-10-09 192115

Comment (seriouslysean) [reply .eslintrc.js:58]: That feels ok with me, but test it and we’ll let @itsalaidbacklife have final say.

Comment (Mang0codes) [reply .eslintrc.js:59]: > That feels ok with me, but test it and we’ll let @itsalaidbacklife have final say. Seems like a bit of research is needed first.

Is this okay? @seriouslysean @itsalaidbacklife

Comment (itsalaidbacklife) [reply .eslintrc.js:59]: Yes I think that's what we want. We'll have to see how it goes and it's plausible that we may need to tweak or backpedal at that point but let's try it and see

Comment (Mang0codes) [reply .eslintrc.js:59]: I've already committed the changes... You may review them

File: api/controllers/game/move.js [hunk 3]
```diff
@@ -13,14 +13,14 @@ module.exports = async function (req, res) {
...
-      default:
-        throw new Error('You are not a player in this game!');
+    case game.p0.id:
+      playedBy = 0;
+      break;
+    case game.p1.id:
+      playedBy = 1;
+      break;
+    default:
+      throw new Error('You are not a player in this game!');
```
Comment (itsalaidbacklife) [inline api/controllers/game/move.js:23]: I'd like to preserve indenting one level for switch cases, so I believe we need to set `SwitchCase: 1` on the indent rule in .eslintrc

File: .eslintrc.js (hunk 2 above)
Comment (itsalaidbacklife) [reply .eslintrc.js:59]: Let's additionally configure the SwitchCase option to preserve the single level of indentation applied to switch cases. It'll mean using the object definition of the rule and setting 

`SwitchCase: 1`

**Review by itsalaidbacklife — COMMENTED** [review]
This looks good, but let's preserve the existing switch case indentation using the SwitchCase option of the indent rule

**Comment by itsalaidbacklife** [comment]
@Mang0codes did you mean to close this? I see that there was another force push and the PR closed

**Comment by Mang0codes** [comment]
> @Mang0codes did you mean to close this? I see that there was another force push and the PR closed

I didn't mean to, I guess this occured due to the git reset command. I'll go ahead and create another pull request including the updates you mentioned.

**Pull request closed without merging**